src/main/java/org/sunbird/ratings/service/RatingServiceImpl.java

```diff
@@ -76,28 +74,89 @@ public SBApiResponse getRatings(String activity_Id, String activity_Type, String
 
             validateRatingsInfo(validationBody, "getSummary");
 
             Map<String, Object> request = new HashMap<>();
-            request.put(Constants.ACTIVITY_ID, activity_Id);
-            request.put(Constants.ACTIVITY_TYPE, activity_Type);
+            request.put(Constants.ACTIVITY_ID, activityId);
+            request.put(Constants.ACTIVITY_TYPE, activityType);
 
             List<Map<String, Object>> existingDataList = cassandraOperation.getRecordsByProperties(Constants.KEYSPACE_SUNBIRD,
                     Constants.TABLE_RATINGS_SUMMARY, request, null);
+
             if (!CollectionUtils.isEmpty(existingDataList)) {
+            Map<String, Object> map = existingDataList.get(0);
+            String reviews = (String) map.get(Constants.LATEST50REVIEWS);
+            JsonNode actualObj = mapper.readTree(reviews);
+            List<String> userList = new ArrayList<>();
+            Map<String, SummaryNodeModel> reviewMap = new HashMap<>();
+
+            for (JsonNode jsonNode : actualObj) {
+                final SummaryNodeModel summaryModel = mapper.convertValue(jsonNode, SummaryNodeModel.class);
+                reviewMap.put(summaryModel.getUser_id(), summaryModel);
+                userList.add(jsonNode.get("user_id").asText());
+            }
+
+            Map<String, Object> userRequest = new HashMap<>();
+            userRequest.put(Constants.USERID, userList);
+            List<String> fields = new ArrayList<>();
+            fields.add(Constants.USERID);
+            fields.add(Constants.FIRSTNAME);
+            fields.add(Constants.LASTNAME);
+
+            Map<String, Object> existingUserList = cassandraOperation.getRecordsByProperties(Constants.KEYSPACE_SUNBIRD,
+                    Constants.TABLE_USER, userRequest, fields, "id");
+            List<SummaryModel.latestReviews> latest50Reviews = new ArrayList<>();
+
+            String firstName = "";
+            String lastName = "";
+
+            for (String user : userList) {
+                final ObjectMapper mapper = new ObjectMapper();
+                final UserModel userModel = mapper.convertValue(existingUserList.get(user), UserModel.class);
+                final SummaryNodeModel summaryNodeModel = mapper.convertValue(reviewMap.get(user), SummaryNodeModel.class);
+                if (userModel.getFirstName() != null) {
+                    firstName = userModel.getFirstName();
+                }
+                if (userModel.getLastName() != null) {
+                    lastName = userModel.getLastName();
+                }
+                latest50Reviews.add(new SummaryModel.latestReviews(Constants.REVIEW,
+                        userModel.getId(),
+                        summaryNodeModel.getDate(),
+                        summaryNodeModel.getRating().floatValue(),
+                        summaryNodeModel.getReview(),
+                        firstName,
+                        lastName
```

Comment:
variable will hold the previous iteration values if null... avoid variable and directly pass the value by checking condition